src/zenml/models/v2/base/base.py

```diff
@@ -255,15 +256,35 @@ def get_body(self) -> AnyBody:
 
         Returns:
             The body field of the response.
+
+        Raises:
+            IllegalOperationError: If the user lacks permission to access the
+                entity represented by this response.
         """
+        if not self.body:
```

Comment:
This same error message is now raised if by some mistake, the body was not populated. Does it make sense to check if `permission_denied` is True and only then raise the permission exception if then, but raise some other exception if `permission_denied` is False ?

Reply:
Done